File: call/src/main/java/com/like/chengdu/call/java/AudioConverter.java
package com.like.chengdu.call.java;

import com.mobile.ffmpeg.FFmpeg;

import java.io.File;
import java.util.Locale;

/**
 * 音频格式转换工具类
 */
public class AudioConverter {
    public static File convertToWav(File file) {
        if (file == null) {
            return null;
        }
        String filePath = file.getPath();
        // 录音文件格式作最好都转为：mp3,  wav
        if (filePath.endsWith(".mp3") || filePath.endsWith(".wav")) {
            return file;
        }
        return convert(file, "wav");
    }

    private static File convert(File file, String format) {
        if (!file.exists() || !file.canRead()) {
            return file;
        }
        File convertedFile = replaceSuffix(file, format);
        String[] cmd = new String[]{"-y", "-i", file.getPath(), convertedFile.getPath()};
        int result = -1;
        try {
            result = FFmpeg.execute(cmd);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return result == 0 ? convertedFile : file;
    }

    private static File replaceSuffix(File originalFile, String format) {
        String[] f = originalFile.getPath().split(".");
        String filePath = originalFile.getPath().replace(f[f.length - 1], format.toLowerCase(Locale.ROOT));
        return new File(filePath);
    }
}

File: call/src/main/java/com/like/chengdu/call/AudioConverter.kt
package com.like.chengdu.call

import com.mobile.ffmpeg.FFmpeg
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import java.io.File
import java.util.*

/**
 * 音频格式转换工具类
 */
object AudioConverter {
    suspend fun convertToWav(file: File?): File? {
        file ?: return null
        val extension = file.extension
        // 录音文件格式作最好都转为：mp3,  wav
        if (extension == "mp3" || extension == "wav") {
            return file
        }
        return convert(file, "wav")
    }

    private suspend fun convert(file: File, format: String): File = withContext(Dispatchers.IO) {
        if (!file.exists() || !file.canRead()) {
            return@withContext file
        }
        val convertedFile = replaceSuffix(file, format)
        val cmd = arrayOf("-y", "-i", file.path, convertedFile.path)
        val result = try {
            FFmpeg.execute(cmd)
        } catch (e: Exception) {
            -1
        }
        if (result == 0) convertedFile else file
    }

    private fun replaceSuffix(originalFile: File, format: String): File {
        val f = originalFile.path.split(".").toTypedArray()
        val filePath =
            originalFile.path.replace(f[f.size - 1], format.lowercase(Locale.getDefault()))
        return File(filePath)
    }

}
